nio-impl/src/main/java/org/xnio/nio/Log.java

```diff
@@ -120,6 +120,8 @@ interface Log extends BasicLogger {
     @Message(id = 7007, value = "Thread is terminating")
     RejectedExecutionException threadExiting();
 
+    @Message(id = 7008, value = "%s - failed to connect to %s")
+    IOException connectionRefused(String originalMsg, String address, @Cause Throwable cause);
```

Comment:
I think we've had previous privacy concerns about logging IP addresses, and I believe this would run afoul of that.